runner/src/scenario/ceramic/mod.rs

```diff
@@ -23,9 +26,147 @@ pub struct Credentials {
+    UserDidKey,
+    // Use CACAOs for each user
+    //UserCacao,
+}
+
+#[derive(Debug, Clone, PartialEq, Eq, serde::Deserialize, serde::Serialize)]
+#[serde(rename_all = "camelCase")]
+pub enum ReuseType {
+    /// Create a new model or model instance document for each user
+    PerUser,
+    /// Create a new model for each node (worker)
+    PerNode,
+    /// Reuse the same model or model instance document for all users
+    Shared,
+}
+
+#[derive(Serialize, Deserialize, Debug, PartialEq, Clone)]
+#[serde(rename_all = "camelCase")]
+pub struct CeramicScenarioParameters {
+    pub did_type: DidType,
+    /// Whether models should be shared or independent
+    pub model_reuse: ReuseType,
+    /// How many model instance documents to create in advance for each model.
+    pub model_instance_reuse: ReuseType,
+    pub number_of_documents: usize,
+}
+
+impl From<Scenario> for CeramicScenarioParameters {
+    fn from(value: Scenario) -> Self {
+        // did_type: DidType::UserDidKey and model_instance_reuse: ReuseType::Shared is an invalid combination
+        // as we'll try to update documents owned by another controller and just log lots of errors
+        match value {
+            Scenario::CeramicSimple => Self {
+                did_type: DidType::Shared,
+                model_reuse: ReuseType::PerUser,
+                model_instance_reuse: ReuseType::PerUser,
+                number_of_documents: 1,
+            },
+            Scenario::CeramicModelReuse => Self {
+                did_type: DidType::UserDidKey,
+                model_reuse: ReuseType::Shared,
+                model_instance_reuse: ReuseType::PerUser,
+                number_of_documents: 1,
+            },
+            Scenario::CeramicWriteOnly => Self {
+                did_type: DidType::UserDidKey,
+                model_reuse: ReuseType::Shared,
+                model_instance_reuse: ReuseType::PerUser,
+                number_of_documents: 1,
+            },
+            Scenario::CeramicNewStreams => Self {
+                did_type: DidType::UserDidKey,
+                model_reuse: ReuseType::PerUser,
+                model_instance_reuse: ReuseType::PerUser,
+                number_of_documents: 0,
+            },
+            Scenario::CeramicNewStreamsBenchmark => Self {
+                did_type: DidType::UserDidKey,
+                model_reuse: ReuseType::Shared,
+                model_instance_reuse: ReuseType::PerUser,
+                number_of_documents: 0,
+            },
+            Scenario::CeramicQuery => Self {
+                did_type: DidType::Shared,
+                model_reuse: ReuseType::PerUser,
+                model_instance_reuse: ReuseType::PerUser,
+                number_of_documents: 3,
+            },
+            Scenario::IpfsRpc | Scenario::ReconEventSync | Scenario::ReconEventKeySync => {
+                panic!("Not supported for non ceramic scenarios")
+            }
+        }
+    }
```

Comment:
I love this. Its a great stepping stone towards having a single well parameterized scenario.